data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/configuration/KafkaKeyMode.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.kafka.configuration;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import java.util.Map;
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public enum KafkaKeyMode {
+    DISCARD("discard"),
```

Comment:
Where is this used? I don't see a reference in this package. Is it needed?

Reply:
It's not in the code because it is the default in the logic. I can add a comment if that helps.

Reply:
It's not codified or enforced anywhere it just daed code. Additionally the default logic or behavior changes based on the plaintext mode in the `KafkaSourceCustomConsumer`. I am a little concerned these options will be confusing to users but I don't have enough context here. If no-one else is concerned then I am fine.

Reply:
In the config, the users will have option to use one of the three. So, they have to use DISCARD option in the config to get this behavior. All this behavior will be put in README file with examples.

Reply:
Plaintext config messages either come with a key and value or null key and value. In case of null key, we set it to "message". We need a key and value to put in DataPrepper Event. So, it should not be confusing for users. Also, in most cases, kafka users use either JSON or AVRO.